ui/app/AppLayouts/Wallet/controls/AssetSelector.qml

```diff
@@ -12,31 +12,32 @@ Control {
     id: root
 
     /** Expected model structure: see SearchableAssetsPanel::model **/
-    property alias assetsModel: searchableAssetsPanel.model
+    property alias model: searchableAssetsPanel.model
+    property alias nonInteractiveKey: searchableAssetsPanel.nonInteractiveKey
 
-    readonly property bool isTokenSelected: d.isTokenSelected
+    property alias sectionProperty: searchableAssetsPanel.sectionProperty
 
-    signal assetSelected(string key)
+    readonly property bool isSelected: button.selected
+
+    signal selected(string key)
 
     function setCustom(name: string, icon: url, key: string) {
-        d.isTokenSelected = true
-        tokenSelectorButton.name = name
-        tokenSelectorButton.icon = icon
+        button.name = name
```

Comment:
Isn't it an issue if the custom data is not available in the model?

I'm wondering why we have this `setCustom` instead of selecting just the key from the outside and populate the button data with the data available from the model.

Reply:
This approach gives a bit more flexibility but I'm not fully sure now if it's needed (it's possible to set content not available in model).

At some point it was said that history of activities may refer to tokens that are not currently in the wallet. So if we want to open send/swap modal on it, me must have possibility to set entry not present in the model normally used for selection. But indeed I'm not sure if it's necessary. Anyways, it was motivation of that approach.